### DISCUSSION TITLE
[@types/lodash] Wrong type for isEmpty

### DISCUSSION THREAD

**Discussion by NogaMan** [DefinitelyTyped/DefinitelyTyped#62155] [Issues with a @types package]
Having the following issue:
```ts
type Errors = Record<string, string>

const errors: Errors = {};

if (isEmpty(errors)) {
  errors.test = "123"; // TS2339: Property 'test' does not exist on type 'never'.
}
```

I believe this started to appear after this PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61620
@forivall cc

To me the correct type here should be the type `Record<string, string>` and not `never`

[a comment by an automated account is not shown]

**Comment by aj-r** [comment]
@forivall do you see a way to fix this? Or do we need to revert https://github.com/DefinitelyTyped/DefinitelyTyped/pull/61620?

**Reply by forivall** [reply]
I'll have to investigate. I think I can solve it without reverting

**Reply by aj-r** [reply]
Unfortunately I think this means we need to stop narrowing the type for objects. While we know that the object contains no properties, it should be valid for users to add properties after calling `isEmpty()`. So it looks like we can really only do type narrowing for strings.

It looks like some unexpected type narrowing is happening in the other direction too: #62379.

**Reply by forivall** [reply]
solved: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62409

**Comment by forivall** [comment]
I can absolutely find a way to fix it, I'll take a look at it soon.

~ Emily Klassen
